zipline/pipeline/data/dataset.py

```diff
@@ -115,14 +118,15 @@ class BoundColumn(LoadableTerm):
     inputs = ()
     window_safe = True
 
-    def __new__(cls, dtype, missing_value, dataset, name):
+    def __new__(cls, dtype, missing_value, dataset, name, ndim):
```

Comment:
hmm, maybe then we should just forward `dataset.ndim` on the line below. It seems strange to be able to pass an `ndim` that doesn't match the `ndim` of the dataset. Idk how this plays with the caching here. ping @ssanderson